File: src/main/java/dev/kikugie/techutils/config/malilib/InGameNotifier.java
package dev.kikugie.techutils.config.malilib;

import fi.dy.masa.malilib.gui.Message;
import fi.dy.masa.malilib.gui.interfaces.IMessageConsumer;
import fi.dy.masa.malilib.render.MessageRenderer;

public class InGameNotifier implements IMessageConsumer {
    public static final InGameNotifier INSTANCE = new InGameNotifier();
    private final MessageRenderer messageRenderer = new MessageRenderer(0xDD000000, 0xFF999999);

    @Override
    public void addMessage(Message.MessageType type, String messageKey, Object... args) {
        this.messageRenderer.addMessage(type, 2000, messageKey, args);
    }

    @Override
    public void addMessage(Message.MessageType type, int lifeTime, String messageKey, Object... args) {
        this.messageRenderer.addMessage(type, lifeTime, messageKey, args);
    }
}

File: src/main/kotlin/me/kikugie/techutils/config/InGameNotifier.kt
package me.kikugie.techutils.config

import fi.dy.masa.malilib.gui.Message
import fi.dy.masa.malilib.gui.interfaces.IMessageConsumer
import fi.dy.masa.malilib.render.MessageRenderer

class InGameNotifier : IMessageConsumer {
    override fun addMessage(type: Message.MessageType, messageKey: String, vararg args: Any) {
        messageRenderer.addMessage(type, 2000, messageKey, *args)
    }

    override fun addMessage(type: Message.MessageType, lifeTime: Int, messageKey: String, vararg args: Any) {
        messageRenderer.addMessage(type, lifeTime, messageKey, *args)
    }

    companion object {
        val INSTANCE = InGameNotifier()
        private val messageRenderer = MessageRenderer(-0x23000000, -0x666667)
    }
}